gcp/table_gcp_redis_instance.go

```diff
@@ -0,0 +1,360 @@
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// standard steampipe columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromP(gcpRedisInstanceTurbotData, "Akas"),
+			},
+
+			// standard gcp columns
+			{
+				Name:        "location",
+				Description: ColumnDescriptionLocation,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromConstant("global"),
+			},
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     plugin.HydrateFunc(getProject).WithCache(),
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTIONS
+
+func listGcpRedisInstances(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+
+	// Create Service Connection
+	service, err := RedisService(ctx, d)
+	if err != nil {
+		logger.Error("gcp_redis_instance.listGcpRedisInstances", "connection_error", err)
+		return nil, err
+	}
+
+	var location string
+	matrixLocation := d.EqualsQualString(matrixKeyRedisLocation)
+	// Since, when the service API is disabled, matrixLocation value will be nil
+	if matrixLocation != "" {
+		location = matrixLocation
+	}
+
+	getProjectCached := plugin.HydrateFunc(getProject).WithCache()
+	projectId, err := getProjectCached(ctx, d, h)
+	if err != nil {
+		logger.Error("gcp_redis_instance.listGcpRedisInstances", "cache_error", err)
+		return nil, err
+	}
+	project := projectId.(string)
+
+	parent := "projects/" + project + "/locations/" + location
+	req := &redispb.ListInstancesRequest{
+		Parent: parent,
+	}
+
+	it := service.ListInstances(ctx, req)
+	for {
+		resp, err := it.Next()
+		if err != nil {
+			if err == iterator.Done {
+				break
+			}
+			logger.Error("gcp_redis_instance.listGcpRedisInstances", "api_error", err)
+			return nil, err
+		}
+		test := resp.CreateTime.AsTime()
+		fmt.Println(test)
+		d.StreamListItem(ctx, resp)
+
+		if d.RowsRemaining(ctx) == 0 {
+			return nil, nil
+		}
+	}
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getGcpRedisInstance(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	instanceName := d.EqualsQualString("name")
+	instanceLocationId := d.EqualsQualString("location_id")
+
+	if instanceName == "" || instanceLocationId == "" {
+		return nil, nil
+	}
+
+	// Create Service Connection
+	service, err := RedisService(ctx, d)
+	if err != nil {
+		logger.Error("gcp_redis_instance.getGcpRedisInstance", "connection_error", err)
+		return nil, err
+	}
+
+	var location string
+	matrixLocation := d.EqualsQualString(matrixKeyRedisLocation)
```

Comment:
From the [above code](https://github.com/turbot/steampipe-plugin-gcp/pull/482/files#diff-db75ec1889ca306d635576b73908993313af659ae4c3791d6c152bbb65a3778cR298) I believe we have to specify the `location_id` in where clause. Do we need this block of code to get the location value from the matrix?

Reply:
> From the [above code](https://github.com/turbot/steampipe-plugin-gcp/pull/482/files#diff-db75ec1889ca306d635576b73908993313af659ae4c3791d6c152bbb65a3778cR298) I believe we have to specify the `location_id` in where clause. Do we need this block of code to get the location value from the matrix?

@ParthaI we require this because the response of the list call returns in the form `us-central1-b` where as the getMatrix give as `us-central` so we need to remove the AZ that's why we need this.